### PR TITLE
RFC: etcdserver: swap priority of cert CN and username + password

### PR DESCRIPTION
**Pull request by mitake**
/cc @raoofm 

I'd like to hear comments about we should add an option for controlling the priority or not.

Fix https://github.com/coreos/etcd/issues/8584

### CODE REVIEW

**Comment by xiang90** [comment]
we probably should add a test?

**Comment by mitake** [comment]
@xiang90 sure, I'll add the test later. How do you think about the option? I'm feeling that just mentioning on release not would be enough, but not sure yet.

**Comment by raoofm** [comment]
@mmerrill3 imo a flag is an overkill. What's your take? Please read #8584

**Comment by mmerrill3** [comment]
@raoofm, yeah, its more natural to use the presented username if it is there.

**Comment by xiang90** [comment]
If we have not documented the previous behavior explicitly, I think we should just fix this and start to document it. This behavior is more natural, and the previous behavior seems buggy.

**Comment by xiang90** [comment]
@mitake can you add a test and let us get this merged.

**Comment by mitake** [comment]
@xiang90 added a commit for the test case, could you take a look? The second commit also fixes a silly bug in the existing `TestCtlV3AuthCertCN` (it didn't work as a test...).

**Comment by gyuho** [comment]
lgtm. /cc @xiang90

File: e2e/ctl_v3_auth_test.go [hunk 1]
```diff
@@ -960,3 +964,55 @@ func authTestEndpointHealth(cx ctlCtx) {
 		cx.t.Fatalf("endpointStatusTest ctlV3EndpointHealth error (%v)", err)
 	}
 }
+
+func authTestCertCNAndUsername(cx ctlCtx) {
```
Comment (xiang90) [inline e2e/ctl_v3_auth_test.go]: we should not use t.fatal all over the place.

fatal should be used for the pre-condition check, errorf should be used for the actual correctness checking.

**Comment by xiang90** [comment]
LGTM after fixing the test nits.

File: e2e/ctl_v3_auth_test.go (hunk 1 above)
Comment (mitake) [reply e2e/ctl_v3_auth_test.go]: I see, thanks for pointing out. I'll update soon.

**Comment by mitake** [comment]
@xiang90 updated for the test nits, could you take a look?

**Comment by xiang90** [comment]
lgtm

**Pull request merged**